File: RepoTransBench/target_projects/Python/C++/OilerNetwork_fossil_cairo0/public_tests/test_public_setup_py.cpp
#include <gtest/gtest.h>
#include <fstream>
#include <string>
#include <sys/stat.h>

// Helper function to check file existence using POSIX APIs
bool file_isfile(const std::string& name) {
    struct stat buffer;
    return (stat(name.c_str(), &buffer) == 0 && S_ISREG(buffer.st_mode));
}

// Public variant: test setup.py file exists using isfile instead of exists
TEST(TestPublicSetupPy, PublicSetupPyExists) {
    // Test that setup.py file exists - public variant.
    EXPECT_EQ(file_isfile("setup.py"), true);
}

// "Import" test - check can open setup.py; accept 'scm' or 'build' if present
TEST(TestPublicSetupPy, PublicImports) {
    std::ifstream infile("setup.py");
    ASSERT_TRUE(infile.is_open()) << "Cannot open setup.py for reading (file must exist).";
    try {
        std::string line;
        bool found_scm = false, found_build = false;
        while (std::getline(infile, line)) {
            if (line.find("scm") != std::string::npos ||
                line.find("SCM") != std::string::npos) {
                found_scm = true;
            }
            if (line.find("build") != std::string::npos ||
                line.find("BUILD") != std::string::npos) {
                found_build = true;
            }
        }
        // Accept either build or scm present (simulate error/exception message)
        EXPECT_TRUE(found_build || found_scm);
    } catch (const std::exception& e) {
        std::string what(e.what());
        EXPECT_TRUE(what.find("build") != std::string::npos ||
                    what.find("scm") != std::string::npos);
    }
}

// Metadata field check (public variant)
TEST(TestPublicSetupPy, PublicMetadata) {
    // Test that setup.py includes "install_requires" and "setuptools"
    std::ifstream infile("setup.py");
    ASSERT_TRUE(infile.is_open()) << "Cannot open setup.py for reading (file must exist).";
    std::string content((std::istreambuf_iterator<char>(infile)), std::istreambuf_iterator<char>());
    EXPECT_NE(content.find("install_requires"), std::string::npos);
    EXPECT_NE(content.find("setuptools"), std::string::npos);
}

File: RepoTransBench/target_projects/Python/Rust/OilerNetwork_fossil_cairo0/public_tests/test_public_setup_py.rs
use std::fs;
use std::io::Read;
use std::path::Path;

#[test]
fn test_public_setup_py_exists() {
    // Test that setup.py file exists - public variant.
    // Use is_file instead of exists.
    assert!(
        Path::new("setup.py").is_file(),
        "setup.py file does not exist or is not a file"
    );
}

#[test]
fn test_public_imports() {
    // Test that setup.py can be imported/syntax-checked (excluding SCM/build logic)
    use std::process::Command;

    // Try running a syntax check (equivalent to import test). We use `python3 -m py_compile setup.py`.
    let output = Command::new("python3")
        .arg("-m")
        .arg("py_compile")
        .arg("setup.py")
        .output()
        .expect("failed to execute python3");
    let stderr = String::from_utf8_lossy(&output.stderr).to_lowercase();
    if !output.status.success() {
        // Allow version/build/scm errors but not syntax errors
        assert!(
            stderr.contains("scm") || stderr.contains("build"),
            "Import (py_compile) failed: {}",
            stderr
        );
    }
}

#[test]
fn test_public_metadata() {
    // Test that setup.py includes different expected metadata fields (public variant)
    let mut file = fs::File::open("setup.py").expect("unable to open setup.py");
    let mut content = String::new();
    file.read_to_string(&mut content).expect("unable to read setup.py");
    // Use a different field from the project metadata, such as install_requires and setuptools
    assert!(
        content.contains("install_requires"),
        "setup.py missing 'install_requires' field"
    );
    assert!(
        content.contains("setuptools"),
        "setup.py missing 'setuptools' field"
    );
}